doc/src/BCMPackage.rst

```diff
@@ -74,3 +90,5 @@ The source files to build the library.
 
 This specifies internal boost dependencies, that is, dependencies on other boost libraries. The libraries should not be prefixed with ``boost_`` nor ``boost::``.
 
+.. Why not use 'boost::' here? This special casing will be unobvious for casual
+   readers of code which uses this library.
```

Comment:
This is good point. I didn't because it always needs boost dependencies, so it might be a good idea to always require it to make it explicit that its a boost dependency:

```cmake
bcm_boost_package(boost_core
    VERSION 1.61.0
    DEPENDS
        boost::assert
        boost::config
)
```

I don't know if it always helps clarity as the package is declared with `boost_` and the dependency that is used starts with `boost::`. Maybe it might be better to say `boost_assert` instead of `boost::assert`. I am not sure.